### ISSUE TITLE
Add --pull functionality to make-override [Feature Request]

### ISSUE THREAD

**Issue by jazzace** [autopkg/autopkg#661]
**Describe the problem**
The `--pull` functionality from the `info` verb would be beneficial to add to the `make-override` verb.

**Current Functionality/Workflow**
When a user finds a new recipe that they want to add to their AutoPkg environment, a common workflow is to add the repo for the recipe (if necessary) to their local AutoPkg instance, then `make-override`. Sometimes, the user does not have all the parent repos for the new recipe downloaded, so AutoPkg will report an error that a parent repo is missing. The user then can add the missing repo using `repo-add <repo>` or `info --pull <recipe>`, then try `make-override` again. For example:  
```
% autopkg make-override outset.ds
Didn't find a recipe for com.github.grahamgilbert.download.outset.
Could not find parent recipe for outset.ds
No valid recipe found for outset.ds
Dir(s) searched:
	.
	~/Library/AutoPkg/Recipes
	/Library/AutoPkg/Recipes
	/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes
	/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.jazzace-recipes
	/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.jazzace-recipes/outset
% autopkg info --pull outset.ds
Didn't find a recipe for com.github.grahamgilbert.download.outset.
Found this recipe in repository: grahamgilbert-recipes
Attempting git clone...

Adding /Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.grahamgilbert-recipes to RECIPE_SEARCH_DIRS...
Updated search path:
  '.'
  '~/Library/AutoPkg/Recipes'
  '/Library/AutoPkg/Recipes'
  '/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes'
  '/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.jazzace-recipes'
  '/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.grahamgilbert-recipes'

Description:         Downloads the latest version of outset (pkg installer) and copies the package to a local path of your choosing.
                     Requires Graham Gilbert's recipe repo (grahamgilbert-recipes).
                     
                     Input keys:
                     - DS_PKGS_PATH is the destination path for the copy. A trailing slash is not required/desired.
                       The path does not need to be on the same volume as the cache, as the pkg is being duplicated.
                     - DS_NAME is what the final package will be called, regardless of what was generated by the parent recipe.
                       It defaults to %NAME% (just the app name).
Identifier:          com.github.jazzace.ds.outset
Munki import recipe: False
Has check phase:     True
Builds package:      False
Recipe file path:    /Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.jazzace-recipes/outset/outset.ds.recipe
Parent recipe(s):    /Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.grahamgilbert-recipes/outset/outset.download.recipe
Input values: 
    'DS_NAME': '%NAME%',
    'DS_PKGS_PATH': '/Volumes/DSServer/Users/Shared/Deploy/Packages',
    'NAME': 'outset'
% autopkg make-override outset.ds
Override file saved to /Users/autopkguser/Library/AutoPkg/RecipeOverrides/outset.ds.recipe
```

**Requested Functionality**
It would be very useful if users could simplify this workflow by adding the `--pull` option to the initial `make-override` request, making the subsequent steps unnecessary. Using the same example:
```
% autopkg make-override --pull outset.ds
Didn't find a recipe for com.github.grahamgilbert.download.outset.
Found this recipe in repository: grahamgilbert-recipes
Attempting git clone...

Adding /Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.grahamgilbert-recipes to RECIPE_SEARCH_DIRS...
Updated search path:
  '.'
  '~/Library/AutoPkg/Recipes'
  '/Library/AutoPkg/Recipes'
  '/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes'
  '/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.jazzace-recipes'
  '/Users/autopkguser/Library/AutoPkg/RecipeRepos/com.github.autopkg.grahamgilbert-recipes'
Override file saved to /Users/autopkguser/Library/AutoPkg/RecipeOverrides/outset.ds.recipe
```

Should the new feature be added, it would also be beneficial for the output of the `make-overrides` command (when called without `--pull`) to add a line to the error output pointing to the new feature, such as: `To automatically retrieve the parent repos needed for a recipe, use the --pull option.`

**Comment by grahampugh** [comment]
+1 for this feature. It makes more sense for this functionality to be a part of make-override than the info option. 

An alternative, to keep it part of the setup phase, would be to implement it for repo-add, where you specify a recipe and it adds all the repos required. Something like:

    autopkg repo-add —recipe MyApp.munki

This removes another step where you have to add the repo of the recipe and then the repos of the parents separately. It would have to work with identifiers for the case of recipes with identical names, since search order is not consistent. 

If you were uncomfortable with adding it to repo-add, I would suggest a new verb: add-repos-for-recipe.

**Comment by nmcspadden** [comment]
Implemented in 3245d12b2e1f6b57b5932b3e25ec26fc3119284a.
